src/osd/osd_types.cc

```diff
@@ -561,15 +590,26 @@ void pg_t::generate_test_instances(list<pg_t*>& o)
   o.push_back(new pg_t(131223, 4, 23));
 }
 
-ostream& operator<<(ostream& out, const pg_t &pg)
+char *pg_t::calc_name(char *buf, const char *suffix_backwords) const
 {
-  out << pg.pool() << '.';
-  out << hex << pg.ps() << dec;
+  while(*suffix_backwords)
+    *--buf = *suffix_backwords++;
+
+  if (m_preferred >= 0)
+    *--buf ='p';
```

Comment:
FWIW, separately, we should eliminate teh m_preferred field entirely.. it's been deprecated for a couple years now.  Probably we can just expand the m_seed to a uint64_t, which we may eventually need.